src/module.c

```diff
@@ -13489,6 +13492,99 @@ int RM_GetDbIdFromDefragCtx(RedisModuleDefragCtx *ctx) {
     return ctx->dbid;
 }
 
+/* --------------------------------------------------------------------------
+ * ## Key API for Set type
+ *
+ * See also RM_ValueLength(), which returns the length of a set.
+ * -------------------------------------------------------------------------- */
+
+/* Operate a set key.
+ * REDISMODULE_SET_ADD: Add new elements into a set. If the key is an empty key
+ *                      open for writing, it is created with an empty set key in
+ *                      order to add the elements.
+ * REDISMODULE_SET_REM: Remove the specified element from the set and the key will be
+ *                      removed if has no any element in after remove elements operation.
+ * REDISMODULE_SET_ISMEMBER: Determines if member is a member of the set stored at key .
+ *
+ * * **key**: The set key to be operated.
+ * * **op**: The operation to the set key.
+ * * **flags**: The flag for following arguments. 
+ *      REDISMODULE_SET_CSTRING: The following arguments are null terminated C
+ *                               strings instead of RedisModuleString objects.
+ * * **...**: The actual arguments.
+ *
+ * Return value:
+ *
+ * REDISMODULE_SET_ADD: The number of elements were really added.
+ * REDISMODULE_SET_REM: The number of elements were really removed.
+ * REDISMODULE_SET_ISMEMBER: 1 if all the elements are member of the set. Otherwise return 0.
+ *
+ * If the return value is zero, `errno` is set as follows:
+ *
+ * - EINVAL if key is NULL.
+ * - ENOTSUP if the key is associated with a non Set value.
+ * - EBADF if the key was not opened for writing.
+ * - ENOENT if no elements were counted as described under Return value above.
+ *   This is not actually an error. The return value can be e.g. zero if all elements
+ *   had been in the set key for add operation.
+ */
+int RM_SetOperate(RedisModuleKey *key, RedisModuleSetOperation op, int flags, ...) {
+    va_list ap;
+    RedisModuleString *ele;
+    int count = (op == REDISMODULE_SET_ISMEMBER) ? 1 : 0;
+
+    if (!key) {
+        errno = EINVAL;
+        return 0;
+    } else if (op == REDISMODULE_SET_REM || op == REDISMODULE_SET_ISMEMBER) {
```

Comment:
This `else if` is always entered when op is `REDISMODULE_SET_REM` or `REDISMODULE_SET_ISMEMBER`, so why they appear in the second and thrid `else if`, and similary, `REDISMODULE_SET_ADD` shouldn't appear if the third `else if`.

Reply:
This `else if (op == REDISMODULE_SET_REM || op == REDISMODULE_SET_ISMEMBER)` is used to check the key shall not be empty for remove and ismember operation. Add operation can be empty for creating new key. This `else if (op == REDISMODULE_SET_REM || op == REDISMODULE_SET_ADD)` is used to check the key mode must be write mode.